hist/hist/src/TAxis.cxx

```diff
@@ -915,28 +938,72 @@ void TAxis::SetBinLabel(Int_t bin, const char *label)
 ///  - To retrieve the number of axis labels use TAxis::GetNlabels.
 
 void TAxis::ChangeLabel(Int_t labNum, Double_t labAngle, Double_t labSize,
-                               Int_t labAlign, Int_t labColor, Int_t labFont,
-                               TString labText)
+                        Int_t labAlign, Int_t labColor, Int_t labFont,
+                        const TString &labText)
 {
-   if (!fModLabs) fModLabs = new TList();
-
    // Reset the list of modified labels.
    if (labNum == 0) {
-      delete fModLabs;
-      fModLabs  = 0;
+      SafeDelete(fModLabs);
       return;
    }
 
-   TAxisModLab *ml = new TAxisModLab();
-   ml->SetLabNum(labNum);
+   if (!fModLabs) fModLabs = new TList();
+
+   TAxisModLab *ml = FindModLab(labNum);
+   if (!ml) {
+      ml = new TAxisModLab();
+      ml->SetLabNum(labNum);
+      fModLabs->Add(ml);
+   }
+
    ml->SetAngle(labAngle);
    ml->SetSize(labSize);
    ml->SetAlign(labAlign);
    ml->SetColor(labColor);
    ml->SetFont(labFont);
    ml->SetText(labText);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Define new text attributes for the label value "labValue". It allows to do a
+/// fine tuning of the labels. All the attributes can be changed, even the
+/// label text itself.
+///
+/// \param[in] labValue  Axis value to be changed
+/// \param[in] labAngle  New angle value
+/// \param[in] labSize   New size (0 erase the label)
+/// \param[in] labAlign  New alignment value
+/// \param[in] labColor  New label color
+/// \param[in] labFont   New label font
+/// \param[in] labText   New label text
+///
+///  #### Notes:
+///
+///  - If an attribute should not be changed just give the value "-1".
+///  - If labnum=0 the list of modified labels is reset.
+///  - To erase a label set labSize to 0.
+///  - If labText is not specified or is an empty string, the text label is not changed.
+///  - To retrieve the number of axis labels use TAxis::GetNlabels.
+
+void TAxis::ChangeValueLabel(Double_t labValue, Double_t labAngle, Double_t labSize,
```

Comment:
Why a new name ? you can simply keep the name "ChangeLabel" as it is a different signature.

Reply:
It will be very tiny difference in argument types - `Int_t` and `Double_t`. 
Therefore I prefer to have different names.

Reply:
> Well, that's a real difference

But it is very difficult to explain and control. 
Especially, `TAxis::ChangeLabel(0)` removes all custom labels, but `TAxis::ChangeLabel(0.)` will introduce 
custom label for value `0.`. This may have side effects on already existing user code - which I want to avoid.

> Moreover you may want also in the future to change string labels

It is already implemented. In this case value is just index of the custom label. 
But this is very hypothetical usecase - custom labels one can configure from the beginning properly.

Reply:
I see indeed the 0 case is an issue. What about about `ChangeLabel` and `ChangeLabelValue` in that case ? it makes it more obvious it is the same functionality ?

Reply:
Maybe `ChangeLabelOfValue`? 

Because we changing label of the specific value - or changing value label (as I named it now).

Reply:
`ChangeLabelByValue` is that case.

Reply:
> `ChangeLabelByValue` is that case.

Ok, I will rebase PR using this name

Reply:
I think it is better that the two methods start with the same text `ChangeLabel` to make it clear they are the same and have them together in the doxygen manual.